cmd/spire-agent/cli/run/run.go

```diff
@@ -441,51 +443,61 @@ func validateConfig(c *Config) error {
 	return nil
 }
 
-func warnOnUnknownConfig(c *Config, l logrus.FieldLogger) {
+func checkForUnknownConfig(c *Config, l logrus.FieldLogger) (err error) {
+	detectedUnknown := func(section string, keys []string) {
+		l.WithFields(logrus.Fields{
+			"section": section,
+			"keys":    strings.Join(keys, ","),
+		}).Error("Unknown configuration detected")
```

Comment:
Since this effectively terminates the program, we may use the fatal level?

```suggestion
		}).Fatal("Unknown configuration detected")
```

Reply:
Sure! Updated.